### PR TITLE
Shooter 

### PR DESCRIPTION
**Pull request by DrorMargalit17**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/ShooterSubsysem.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+  public static ShooterSubsysem getInstance(){
+    if (instance == null) {
+      instance = new ShooterSubsysem();
+    }
+    return instance;
+  }
+
+  /** Creates a new ShooterSubsysem. */
+  public ShooterSubsysem() {
```
Comment (ori-coval) [inline src/main/java/frc/robot/subsystems/ShooterSubsysem.java]: constructor should be private

File: src/main/java/frc/robot/commands/ShooterSetSpeedInterpolation.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import frc.robot.subsystems.ShooterSubsysem;
+import frc.robot.subsystems.SwerveSubsystem;
+
+public class ShooterSetSpeedInterpolation extends Command {
+  private final ShooterSubsysem shooterSubsystem = ShooterSubsysem.getInstance();
+  private final SwerveSubsystem swerve = SwerveSubsystem.getInstance();
+
+  /** Creates a new ShooterSetSpeedInterpolation. */
+  public ShooterSetSpeedInterpolation() {
+    this.addRequirements(shooterSubsystem, swerve);
```
Comment (ori-coval) [inline src/main/java/frc/robot/commands/ShooterSetSpeedInterpolation.java]: swerve is not a requirement

**Review by ori-coval — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/ShooterSetSpeedInterpolation.java (hunk 2 above)
Comment (DrorMargalit17) [reply src/main/java/frc/robot/commands/ShooterSetSpeedInterpolation.java]: Done.

File: src/main/java/frc/robot/subsystems/ShooterSubsysem.java (hunk 1 above)
Comment (DrorMargalit17) [reply src/main/java/frc/robot/subsystems/ShooterSubsysem.java]: Done.

File: src/main/java/frc/robot/subsystems/ShooterSubsysem.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    public double getShooterSpeed(){
+        return m_shooterMotor.getVelocity().getValue();
+    }
+
+    public boolean checkIfShooterAtSpeed(){
+      return (Math.abs(m_shooterMotor.getClosedLoopError().getValue()) < MaxError);
+    }
+    
+    public double InterpolationValue(Pose2d pose2d){
+      return InterpolateUtil.interpolate(ShooterInterpolation, vision.DistanceFromTarget(swerve.getPose()));
```
Comment (ori-coval) [inline src/main/java/frc/robot/subsystems/ShooterSubsysem.java]: why are you using `swerve.getPose()`  if you got a pose in the () `InterpolationValue(Pose2d pose2d){`

use the pose you got not the`swerve.getPose()`

**Review by ori-coval — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/ShooterSetSpeed.java
Comment (ori-coval) [inline src/main/java/frc/robot/commands/ShooterSetSpeed.java:1]: use instant command

**Review by ori-coval — APPROVED** [review]

**Pull request merged**